### PR TITLE
Clean up PM_DoRangeAttack

### PR DESCRIPTION
**Pull request by AJenbo**
There is some thing odd about _pAnimFrame, I solved it in a way that looks a bit peculiar to me but other combinations didn't seam to give a match, any ideas?

Also there is still some diff with StartStand() etc where my code has both comparison grouped abouve the code and the original does some negative jumps.

billede

### CODE REVIEW

File: Source/player.cpp [hunk 1]
```diff
@@ -3692,44 +3692,59 @@ int __fastcall PM_DoAttack(int pnum)
...
+			mistype,
+			0,
+			pnum,
+			4,
+			0
+		);
+
+		PlaySfxLoc(PS_BFIRE, plr[pnum].WorldX, plr[pnum].WorldY);
+
+		if ( WeaponDur(pnum, 40) ) {
```
Comment (AJenbo) [inline Source/player.cpp]: pretty sure im issing " && plr[pnum]._pAnimFrame >= plr[pnum]._pAFrames" on this one...

File: Source/player.cpp [hunk 2]
```diff
@@ -3692,44 +3692,59 @@ int __fastcall PM_DoAttack(int pnum)
 }
 // 484368: using guessed type int FriendlyMode;
 
-int __fastcall PM_DoRangeAttack(int pnum)
+BOOL __fastcall PM_DoRangeAttack(int pnum)
```
Comment (squidcc) [inline Source/player.cpp]: Looks like you missed changing the return type in the header.

Comment (AJenbo) [reply Source/player.cpp]: Thanks, fixed

**Comment by seritools** [comment]
Welp, TIL that an `else` can change the output... now binary exact.

File: Source/player.cpp (hunk 1 above)
Comment (seritools) [reply Source/player.cpp]: Letting it fall through to the other check works fine now :)

**Pull request merged**